feat: add specification for sign typed data message

rmw2 opened the pull request:
Add a specification for a typed data (ERC712) signature request message.  The message is a JWT much like our other messages, with the difference that the `typedData` field, (in place of `unsignedClaim`) is structured according to ERC712.  The JWT response also contains an explicit signature field, representing the signature over the `typedData` alone, while the JWT itself is signed over its entire structure, including the issuer and subject fields.

Note: The response still feels a little strange to me, I would love input from @aldigjo @simonas-notcat @pelle --  Is it fine to just wrap the concept of an ERC712 typed data and signature inside a JWT?  Specifically, do we want two signatures on a single JWT, one of them inside of the object itself? Also is there still necessarily a concept of a subject?

rmw2 committed:
feat: add specification for sign typed data message

pelle commented on messages/signtypeddata.md:
```diff
@@ -0,0 +1,113 @@
+[`aud`](https://tools.ietf.org/html/rfc7519#section-4.1.3) | The DID or URL of the audience of the JWT. Our libraries or app will not accept any JWT that has someone else as the audience| no
+[`iat`](https://tools.ietf.org/html/rfc7519#section-4.1.6) | The time of issuance | yes
+[`exp`](https://tools.ietf.org/html/rfc7519#section-4.1.4) | Expiration time of JWT | no
+`type` | MUST have the value `signTypedDataReq` | yes
```
Could we maybe make this smaller. I'm even thinking `eip712Req`.

pelle reviewed (changes requested):
Looks great. One small change I'd like. Can we change it to `EIP712Req`. It would allow us to do support future schemes easier,

rmw2 committed:
doc: change type string for EIP712 request

Also replace all occurrences of ERC712 with EIP712, and remove potentially confusing use of the phrase "verified claim"

pelle commented on messages/signtypeddata.md:
````diff
@@ -0,0 +1,113 @@
+
+```json
+{
+  "type":"verReq",
````
Missed this before. It should be `eip712Req` in the example as well

rmw2 replied:
Good catch, thanks

rmw2 committed:
fix: Change type to 'eip712Req' in example
